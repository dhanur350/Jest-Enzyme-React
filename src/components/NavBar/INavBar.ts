export interface INavBar{
    position: "sidenav" | "navbar";
    logoImage?: string | "";
    userImage?: string;
    to?:string;
    navlinks?:[]
    // position: string; focusedLink: string; handleLinkFocus: (buttonId: string) => void;
    // isActive?: (match: match<{}>, location: Location) => boolean;
}
