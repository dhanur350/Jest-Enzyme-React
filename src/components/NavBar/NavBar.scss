@import "../../styles/core/core-variables";
@import "../../styles/core/core-border";

//custom styles heights and colors for sidebar and navbar
$sidebar-focus-text-color: #fff;
$sidebar-focus-bgcolor:#000;
$sidebar-width: 4.5rem;
$navbar-padding: 0.5rem;
$sidebar-border-radius: 1rem;
$image-border-line :1px solid rgb(192, 190, 190);

.navbar_container {
  width: 100%;
  max-height: 5rem;
  height: 100%;
  align-items: flex-end;
  padding-bottom: $navbar-padding;
}


.navbar {
  display: flex;
  flex-direction: row;
  background-color: #f1f1f1;
  justify-content: space-around;
  height: 4.5rem;
  width: 100%;
  max-width: 35rem;
  padding-top: $padding-2;
  padding-bottom: $padding-2;

  .navbar_link_text {
    display: none;
  }

  a {
    flex-direction: column;
    text-align: center;
    max-width: $sidebar-width;

    .navbar_icons {
      color: gray;
    }
  }

  a:hover {
    align-items: center;
    max-width: $sidebar-width;
    cursor: pointer;
  }

    a:focus,
    a:active,
    a.active {
    align-items: center;
    justify-content: space-evenly;
    flex-direction: row;
    transition: 0.2s;
    width: 100%;
    max-width: 8rem;
    border-radius: $sidebar-border-radius;
    color: $sidebar-focus-text-color;
    background-color: $sidebar-focus-bgcolor;

    .navbar_icons {
      visibility: visible;
      color: white;
    }
    
    .navbar_link_text {
      display: inline;
      background-color: $sidebar-focus-bgcolor;
      color: $sidebar-focus-text-color;
    }

  .navbar_link_text:hover{
    // color: blue;
    display: inline;
    visibility: visible;
  }

  
  }

  .icon_container {
    padding-top: $padding-2;
    padding-bottom: $padding-2;
  }

  .navbar_top_link:hover,
  .navbar_top_link:active,
  .navbar_top_link.active {
    border-radius: $sidebar-border-radius;
    // border-bottom-left-radius: $sidebar-border-radius;
  }

  .navbar_middle_link:hover, 
    .navbar_middle_link:active, 
    .navbar_middle_link.active{
    border-radius: $sidebar-border-radius;
  }

  .navbar_bottom_link:hover, 
    .navbar_bottom_link:active, 
    .navbar_bottom_link.active {
    border-radius: $sidebar-border-radius;
    // border-bottom-right-radius: $sidebar-border-radius;
  }
}

.focused .navbar_link_text:focus {
  visibility: visible;
  // font-weight: bold;
}

.logoimg {
  border-right: $image-border-line;
}

.userlogo {
  border-left: $image-border-line;
}

.logoimg img {
  height: 3rem;
}

.userlogo img {
  height: 3rem;
  border-radius: 100%;
}
