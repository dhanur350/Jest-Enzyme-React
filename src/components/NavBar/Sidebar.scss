$navbar-padding: 0.5rem;
$sidebar-width: 4.5rem;
$sidebar-focus-text-color: #fff;
$sidebar-focus-bgcolor:#000;
$sidebar-border-radius: 1rem;


.sidenav_container {
    flex-direction: column;
    align-items: flex-start;
    padding-left: $navbar-padding;
    max-width: $sidebar-width;
    height: calc(100vh - 6rem);
  }
  
  .sidenav {
    align-items: center;
    justify-content: space-between;
    flex-direction: column;
    max-width: $sidebar-width;
    background-color: #f1f1f1;

    a{
        align-items: center;
    }
  
    a:hover {
      display: flex;
      align-items: center;
      flex-direction: column;
      justify-content: center;
      max-width: $sidebar-width;
      color: blue;
      background-color: lightgray;
    }
  
    a:focus,
      a:active,
      a.active  {
      display: flex;
      align-items: center;
      flex-direction: column;
      max-width: $sidebar-width;
      color: $sidebar-focus-text-color;
      background-color: $sidebar-focus-bgcolor;
    }
  
    a.flex:nth-child(1):active,
  a.flex:nth-child(1):focus,
  a.flex:nth-child(1):hover,
  a.flex:nth-child(1).active {
    border-radius: 1rem 1rem 0 0;
  }

  a.flex:nth-child(3):active,
  a.flex:nth-child(3):focus,
  a.flex:nth-child(3):hover,
  a.flex:nth-child(3).active {
    border-radius: 0 0 1rem 1rem;
  }
  
    .sidenav_top_link:focus {
      border-top-left-radius: $sidebar-border-radius;
      border-top-right-radius: $sidebar-border-radius;
    }
  
    .sidenav_bottom_link:focus {
      border-bottom-left-radius: $sidebar-border-radius;
      border-bottom-right-radius: $sidebar-border-radius;
    }
  }