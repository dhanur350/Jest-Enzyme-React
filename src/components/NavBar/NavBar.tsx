import { memo } from "react";
import { NavLink } from "react-router-dom";
import { Icon } from "semantic-ui-react"
import "./NavBar.scss"
import "./Sidebar.scss"

export interface ILinks {
    label: string;
    to: string;
    icon: any;
    index: string;
    linkText: string;
}


// export interface INavBarMap {
//     position: string ;
//     logoImage?: string;
//     userImage?: string;
//     focusedLink: string;
//     to?: string;
//     handleLinkFocus: (buttonId: string) => void;
// }
export interface INavBarMap {
    position: string;
    logoImage?: string;
    userImage?: string;
    to?: string;
    navLinks?:ILinks[]
}

function NavbarMap({ position, logoImage, userImage,navLinks }: INavBarMap) {
    // console.log(links);

    // useMemo(() => {
    
    // }, [position])


    const renderNavBar = ({ label, icon, to, index }: ILinks) => {
        console.log(label, icon, to, index);
        return (
            <NavLink
                key={`navbar-no.${index}`}
                className={`flex flex-row flex-justify-center flex-column text-secondary-color padding-3`}
                to={to}
            >
                <div className="flex flex-justify-center">
                    <Icon size="big" className="navbar_icons" name={icon} />
                </div>
                <div className={`${position}_link_text `}>{label}</div>
            </NavLink>
        )
    }

    return (
        <nav className={`${position}_container text-sm flex flex-justify-center`}>
            <div className={`${position} box-shadow-navbar border-radius-1`}>
                {logoImage ? (
                    <div className="logoimg flex flex-align-center padding-r-3">
                        <img src={logoImage} alt="Logo" />
                    </div>
                ) : null}
                {navLinks?.map(renderNavBar)}
                {userImage ? (
                    <div className="userlogo flex flex-align-center padding-l-3">
                        <img src={userImage} alt="Logo" />
                    </div>
                ) : null}
            </div>
        </nav>
    )

}

export default memo(NavbarMap)