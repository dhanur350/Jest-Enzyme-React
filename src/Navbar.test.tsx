import { shallow } from "enzyme";
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import NavBar from "./components/NavBar/NavBar";

Enzyme.configure({ adapter: new Adapter() });


const navLinks = [
    {
        index: "1",
        label: "Accounts",
        linkText: "ACCOUNTS",
        to: "/accounts",
        icon: "address card outline",
    },
    {
        index: "2",
        linkText: "PROJECTS",
        label: "Projects",
        to: "/projects",
        icon: "briefcase",
    },
    {
        index: "3",
        linkText: "DIRECTORY",
        label: "Directory",
        to: "/directory",
        icon: "file alternate",
    },
];


describe("Navbar testing", () => {
    it("Test navbar's click", () => {
        const wrapper = shallow(<NavBar position={"navbar"} navLinks={navLinks} />);
        expect(wrapper).toBeTruthy();
    })
});

describe('Navbar Snapshot!', () => {
    it('should match snapshot', () => {
        const component = renderer.create(<NavBar position="sidenav" />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});