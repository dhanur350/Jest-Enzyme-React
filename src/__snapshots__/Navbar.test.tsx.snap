// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should match snapshot 1`] = `
<nav
  className="sidenav_container text-sm flex flex-justify-center"
>
  <div
    className="sidenav box-shadow-navbar border-radius-1"
  />
</nav>
`;

exports[`Navbar Snapshot! should match snapshot 1`] = `
<nav
  className="sidenav_container text-sm flex flex-justify-center"
>
  <div
    className="sidenav box-shadow-navbar border-radius-1"
  />
</nav>
`;
